pkg/controller/common/common.go

```diff
@@ -123,18 +123,46 @@ func (controllerContext ControllerContext) Stop() {
 	controllerContext.Broadcaster.Shutdown()
 }
 
-type ChildMap map[string]map[string]*unstructured.Unstructured
+type GroupVersionKind struct {
+	schema.GroupVersionKind
+}
+
+func (gvk GroupVersionKind) MarshalText() ([]byte, error) {
+	var marshalledText string
+	if gvk.Group == "" {
+		marshalledText = fmt.Sprintf("%s.%s", gvk.Kind, gvk.Version)
+	} else {
+		marshalledText = fmt.Sprintf("%s.%s/%s", gvk.Kind, gvk.Group, gvk.Version)
+	}
+	return []byte(marshalledText), nil
+}
+
+func (gvk *GroupVersionKind) UnmarshalText(text []byte) error {
+	kindGroupVersionString := string(text)
+	parts := strings.SplitN(kindGroupVersionString, ".", 2)
+	groupVersion, err := schema.ParseGroupVersion(parts[1])
+	if err != nil {
+		return err
+	}
+	*gvk = GroupVersionKind{
+		groupVersion.WithKind(parts[0]),
+	}
+	return nil
+}
 
-func (m ChildMap) InitGroup(apiVersion, kind string) {
-	m[childMapKey(apiVersion, kind)] = make(map[string]*unstructured.Unstructured)
+type ChildMap map[GroupVersionKind]map[string]*unstructured.Unstructured
+
+func (m ChildMap) InitGroup(gvk schema.GroupVersionKind) {
+	internalGvk := GroupVersionKind{gvk}
+	m[internalGvk] = make(map[string]*unstructured.Unstructured)
```

Comment:
should we check the presence of old/existing value?
should we set the old value to nil if exists before initialising?

Reply:
I would say it is used in code before inserting objects to newly created child map, so main purpose it to make sure that there is structure initialized. I added check to do nothing if entry is already initialized